src/main/java/edu/ucla/library/libcal/handlers/ProxyHandler.java

```diff
@@ -0,0 +1,133 @@
+
+package edu.ucla.library.libcal.handlers;
+
+import static edu.ucla.library.libcal.MediaType.APPLICATION_JSON;
+import static edu.ucla.library.libcal.MediaType.TEXT_PLAIN;
+
+import info.freelibrary.util.HTTP;
+
+import edu.ucla.library.libcal.Constants;
+import edu.ucla.library.libcal.JsonKeys;
+import edu.ucla.library.libcal.MessageCodes;
+import edu.ucla.library.libcal.services.LibCalProxyService;
+import edu.ucla.library.libcal.services.OAuthTokenService;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.MessageConsumer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.serviceproxy.ServiceBinder;
+
+/**
+ * A handler that processes status information requests.
+ */
+@SuppressWarnings("PMD.UnusedLocalVariable")
+public class ProxyHandler implements Handler<RoutingContext> {
+
+    /**
+     * The handler's logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProxyHandler.class, MessageCodes.BUNDLE);
+
+    /**
+     * The handler's copy of the Vert.x instance.
+     */
+    private final Vertx myVertx;
+
+    /**
+     * The handler's copy of the application config.
+     */
+    private final JsonObject myConfig;
+
+    /**
+     * A service for LibCal API calls
+     */
+    private LibCalProxyService myApiProxy;
+
+    /**
+     * A service for LibCal OAuth calls
+     */
+    private OAuthTokenService myTokenProxy;
+
+    /**
+     * Creates a handler that returns a status response.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aConfig A configuration
+     */
+    public ProxyHandler(final Vertx aVertx, final JsonObject aConfig) {
+        myVertx = aVertx;
+        myConfig = aConfig;
+    }
+
+    @Override
+    public void handle(final RoutingContext aContext) {
+        final HttpServerResponse response = aContext.response();
+        final String receivedQuery = aContext.pathParam(Constants.QUERY_PARAM);
+
+        if (receivedQuery == null || receivedQuery.equals(Constants.EMPTY)) {
+            response.setStatusCode(HTTP.BAD_REQUEST).putHeader(HttpHeaders.CONTENT_TYPE, TEXT_PLAIN.toString())
+                    .end(LOGGER.getMessage(MessageCodes.LCP_006));
+            return;
```

Comment:
Is this `return` required here? Wouldn't it just return naturally since nothing else happens in this path before the method ends?

Reply:
Removed